### PR TITLE
WT-7987 Write a new test to verify compact operation in presence of overflow values.

### PR DESCRIPTION
**Pull request by hbokhari**

### CODE REVIEW

File: test/suite/test_compact03.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        # Insert around ~50% of the normal values in the table that we deleted earlier.
+        c = self.session.open_cursor(self.uri, None)
+        for i in range((self.nrecords // 100) * 25, (self.nrecords // 100) * 75):
+            c.set_key(i)
+            c.set_value(self.normalValue)
+            self.assertEqual(c.update(),0)
+        c.close()
+
+        # Perform checkpoint and compact.
+        self.session.checkpoint()
```
[the remark this answers is not in the document]
Comment (hbokhari) [reply test/suite/test_compact03.py]: Fixed in e89a1fa.

**Review by AlexeyAnisimovAU — APPROVED** [review]
LGTM!

File: test/suite/test_compact03.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        # 4. Perform checkpoint to ensure overflow values are written to the disk.
+        self.session.checkpoint()
+        sizeWithOverflow = self.getSize()
+        self.pr('After inserting overflow values ' + str(sizeWithoutOverflow // mb) + 'MB')
+        self.assertGreater(sizeWithOverflow, sizeWithoutOverflow)
+
+        # 5. Delete middle ~90% of the normal values in the table.
+        c = self.session.open_cursor(self.uri, None)
+        for i in range((self.nrecords // 100) * 5, (self.nrecords // 100) * 95):
```
Comment (raviprakashgiri29) [inline test/suite/test_compact03.py:120]: Can you do a fast truncate here with the start and end key values?

Comment (hbokhari) [reply test/suite/test_compact03.py]: I have added a scenario where we use truncate to delete keys. See 0ac13fe.

**Review by raviprakashgiri29 — APPROVED** [review]
LGTM

**Pull request merged**